akka-stream-tests/src/test/scala/akka/stream/io/TcpSpec.scala

```diff
@@ -380,6 +381,23 @@ class TcpSpec extends AkkaSpec("akka.io.tcp.windows-connection-abort-workaround-
       binding.map(_.unbind())
     }
 
+    "handle when connection actor terminates unexpectedly" in {
+      import system.dispatcher
+
+      val serverAddress = temporaryServerAddress()
+      val binding = Tcp().bindAndHandle(Flow[ByteString], serverAddress.getHostName, serverAddress.getPort)
+
+      val result = Source.lazyEmpty[ByteString].via(Tcp().outgoingConnection(serverAddress)).runFold(0)(_ + _.size)
+
+      // Getting rid of existing connection actors by using a blunt instrument
+      system.actorSelection(akka.io.Tcp(system).getManager.path / "selectors" / "$a" / "*") ! Kill
```

Comment:
Ah, forgot to say: this should be done in a separate ActorSystem if we ever run test cases in parallel.